modules/security/src/main/Ip2Proxy.scala

```diff
@@ -4,18 +4,39 @@ import com.github.blemale.scaffeine.AsyncLoadingCache
 import play.api.libs.json._
 import play.api.libs.ws.JsonBodyReadables._
 import play.api.libs.ws.StandaloneWSClient
+import scala.concurrent._
 import scala.concurrent.duration._
 
 import lila.common.IpAddress
 
-final class Ip2Proxy(
+trait Ip2Proxy {
+
+  def apply(ip: IpAddress): Fu[Boolean]
+
+  def keepProxies(ips: Seq[IpAddress]): Fu[Set[IpAddress]]
+}
+
+final class Ip2ProxySkip(
+    ws: StandaloneWSClient,
+    cacheApi: lila.memo.CacheApi,
+) extends Ip2Proxy {
+
+  def apply(ip: IpAddress): Fu[Boolean] = Future.successful(false)
+
+  def keepProxies(ips: Seq[IpAddress]): Fu[Set[IpAddress]] = {
+    val s: Set[IpAddress] = Set()
+    Future(s)(ExecutionContext.global)
```

Comment:
never reference the global EC like this. Look at how other classes get an instance of an EC.

Reply:
:100